src/main/java/step3/model/LottoMachine.java

```diff
@@ -1,50 +1,103 @@
+
+    private LottoElement bonusNumber;
+    private LottoTicket winnerLotto;
 
+    public LottoMachine() {
 
-    public LottoMachine(LottoManager lottoManager) {
-        this.lottoManager = lottoManager;
     }
 
 
-    public HashMap<String, Integer> checkWin(LottoTicket winnerLotto) {
-        return lottoManager.checkWin(winnerLotto);
+    public List<LottoTicket> makeRandomLottoTickets(Ticket ticket) {
+        List<LottoTicket> lottoTickets = new ArrayList<>();
+        for (int i = 0; i < ticket.getTicket(); i++) {
+            lottoTickets.add(makeRandomLottoTicket());
+        }
+        return lottoTickets;
+    }
+
+    private LottoTicket makeRandomLottoTicket() {
+        List<Integer> lottoElements = new ArrayList<>();
+        shuffle(LOTTO_VALID_ELEMENTS);
+        for (int i = 0; i < LOTTO_ELEMENTS_SIZE; i++) {
+            lottoElements.add(LOTTO_VALID_ELEMENTS.get(i));
+        }
+        return new LottoTicket(lottoElements.stream().map(String::valueOf).collect(Collectors.toList()));
+    }
+
+    public boolean setBonusNumber(String lottoElementSource) {
+        if (!validBonusNumber(lottoElementSource)) {
+            System.out.println(SET_BONUS_NUMBER_EXCEPTION_MSG);
+            return false;
+        }
+        bonusNumber = new LottoElement(Integer.parseInt(lottoElementSource));
+        return true;
     }
 
-    public LottoTicket makeManualLottoTicket(String manualLottoSource) {
+    private boolean validBonusNumber(String lottoElementSource) {
+        int isExist = 1;
         try {
-            return lottoManager.makeManualLottoTicket(manualLottoSource);
+            int bonusExistInWinnerTicket = winnerLotto.getMatchCountWith(Arrays.asList(new LottoElement(Integer.parseInt(lottoElementSource))));
+            return bonusExistInWinnerTicket != isExist;
+        } catch (NumberFormatException e) {
+            return false;
         } catch (IllegalArgumentException e) {
-            System.out.println(e.getMessage());
-            return null;
+            return false;
         }
     }
 
-    public int buyTicket(Money money) {
-        return lottoManager.buyRandomTicket(money.purchaseTicket());
+    public boolean setWinnerLotto(String winnerSource) {
+        winnerLotto = makeManualLottoTicket(winnerSource);
+        return winnerLotto != null;
     }
 
-    public Money createMoney(String money) {
+    private LottoTicket makeManualLottoTicket(String manualLottoSource) {
         try {
-            return new Money(money);
+            return new LottoTicket(manualLottoSource);
         } catch (IllegalArgumentException e) {
             System.out.println(e.getMessage());
             return null;
         }
     }
 
-    public List<List<LottoElement>> getLottoNumbers() {
-        return lottoManager.getLottoNumbers();
+    public HashMap<LottoReward, Integer> checkWin(List<LottoTicket> lottoTickets) {
+        LinkedHashMap<LottoReward, Integer> statistics = new LinkedHashMap<>();
```

Comment:
다형성을 위해 다음과 같이 수정하면 좋을 것 같아요.
```suggestion
       Map<LottoReward, Integer> statistics = new LinkedHashMap<>();
```